app/helpers/attendances_helper.rb

```diff
@@ -12,7 +12,8 @@ def reports_params(params)
     reports_values = { day: DateTime.now, employee_id: employee }
   end
 
-  def get_average(average, flag)
-    Attendance.average_check_time_by_month(@avg_month, flag)
+  def get_average(flag)
+    average = Attendance.average_check_time_by_month(@avg_month, flag)
+    average.is_a?(String) ? average : 'Without checks'
```

Comment:
it always helps when you have a (flag) true/false to split the method in two methods:

when you are in the view you'll use this method something like: 

`get_average(true)` and `get_average(false)` 

^ that doesn't tell anything, it's better to be explicit and clear... something more like having two methods: `get_average_without_checks` and/or `get_average` 

or even better:

I can see you are using `Attendance.average_check_time_by_month(@avg_month, flag)` 

https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-f22-t4/blob/5152a60a2973c80d596b98c923e5fe5467997490/app/models/attendance.rb#L17-L24

and I notice you name the variable here `flag` but in the model it is named `check` that's confusing: 

So probably this could benefit from splitting as well this method into two, something more like: 

`Attendance.average_check_in_time_by_month(date)` 
and
`Attendance.average_check_out_time_by_month(date)` 

^ in that case you'll separate the logic and then later if the future if you come back to this code, it will be simpler to read and understand because it only does one thing and it does it well. :) Also always add Unit Testing or Spec ... in that way you ensure nobody breaks it.

Reply:
@vyk2rr, Actually, in the beginning, I had `average_check_in_time_by_month(date)` and `average_check_out_time_by_month(date)`, and my code looked like this:
```ruby
def self.average_check_in_time_by_month(date)  
    attendances = month_attendances(date)
    return attendances unless attendances.length.positive?

    check_list = attendances.map { |attendance| attendance.check_in }
    check_list.reject!(&:nil?)
    Time.at(average(check_list)).strftime('%k:%M')
  end

def self.average_check_out_time_by_month(date)
    attendances = month_attendances(date)
    return attendances unless attendances.length.positive?

    check_list = attendances.map { |attendance| attendance.check_out }
    check_list.reject!(&:nil?)
    Time.at(average(check_list)).strftime('%k:%M')
  end
```
with this the code was clearer, but the two methods were practically the same, the only change was: `attendance.check_out` and `attendance.check_out`. That's why I decided to combine both methods into one and use the boolean value to decide which average to calculate.

Reply:
I made some changes, I correct the boolean varible name in `average_check_time_by_month` and `get_average`, now is called `check_flag`. Also in `average_check_time_by_month` I added a comment explanning the funtion of the `check_flag` variable  to make the method more clear.

Reply:
But I still have a doubt, what was the better option in cases like this... Duplicate some code to make it more clear or avoid duplication but lose some clarity